packages/@angular/cli/utilities/check-port.ts

```diff
@@ -0,0 +1,24 @@
+import * as denodeify from 'denodeify';
+
+const SilentError = require('silent-error');
+const PortFinder = require('portfinder');
+const getPort = <any>denodeify(PortFinder.getPort);
+
+PortFinder.basePort = 49152;
+
+
+export function checkPort(port: number, host: string) {
+  return getPort({ port, host })
+    .then((foundPort: number) => {
+
+      // If the port isn't available and we weren't looking for any port, throw error.
+      if (port !== foundPort && port !== 0) {
```

Comment:
I wonder why you did not just return the found port instead of throwing an error. This would help developers in case the port is already in use, I work with multiple projects at the same time and often find my self having to serve with a random port number. For example `ng serve --port=2010`